### DISCUSSION TITLE
Aqualink unlock

### DISCUSSION THREAD

**Discussion by jsrichie** [aqualinkd/AqualinkD#218] [Ideas]
So I have locked my SWG out but not seeing the aquaria AL-0 setting after replacing the GXL-PCB-DSP board. Is there a way to unlock it? In short, when the Aqualink sees a non-Aqua pure device, it will write 0x12345678 to 4-byte locations in the higher side of the memory, which will lock it out. Reading back the memory from the device will show you and allow you to clear it by overwriting these locations with 0x00000000 (the rest in that area is empty (0x00) anyway). 

Can the AqualinkD run this write as an option or sub-routine for those of us that have locked our systems to any SWG?

**Comment by sfeakes** [comment]
AqualinkD does not support this functionality.   It is a little more complicated to unlock than what’s described above.  I did look at doing this a long time ago, but I don’t have enough equipment / variety of control panels at different Rev levels to fully test adding this functionality in AqualinkD. And If if anything goes wrong it will brick the control panel, costing a lot to either replace or re-flash. So I am apprehensive to add this functionality.
  
As I’m sure you are aware their is this device that will do exactly what you want.  http://www.chlorinatortranslator.com/  This has had a ton of testing, forces a better RS485 connection so less lightly to have errors.

Or, you can use the Jandy simulator tools to download the flash, modify it, and re-upload it. The forum post that looks to be where the above information came from, is also suggesting this.

When I get some time I’ll relook at it / decision I made above.
